docs/learn-glossary.md

```diff
@@ -3,160 +3,175 @@ id: learn-glossary
 
 ### Activity
 
-A business-level function that implements your application logic such as calling
-a service or transcoding a media file. An activity usually implements a single
-well-defined action; it can be short or long running. An activity can be implemented
-as a synchronous method or fully asynchronously involving multiple processes.
-An activity can be retried indefinitely according to the provided exponential retry policy.
-If for any reason an activity is not completed within the specified timeout, an error is reported to the workflow and the workflow decides how to handle it. There is no limit on potential activity
-duration.
+An Activity is a business-level function that implements your application logic, such as calling a service or transcoding a media file.
 
-### Activity Task
+- An Activity usually implements a single well-defined action; it can be short or long running.
+- An Activity can be implemented as a synchronous method or fully asynchronously involving multiple processes.
+- An Activity can be retried indefinitely according to the provided exponential retry policy.
+- If for any reason an Activity is not completed within the specified timeout, an error is reported to the [Workflow](#workflow) which decides how to handle it. There is no limit for an Activity duration.
 
-A task that contains an activity invocation information that is delivered to an [activity worker](#activity-worker) through and an [activity task queue](#activity-task-queue). An activity worker upon receiving activity task executes a correponding [activity](#activity)
+### Acitity ID
 
-### Activity Task Queue
+An Activity ID is a UUID that corresponds to an Activity.
 
-Task queue that is used to deliver [activity tasks](#activity-task) to [activity workers](#activity-worker)
+### Activity Task
 
-### Activity Worker
+A Task that contains invocation information for an [Activity](#activity) that is delivered to an [Activity Worker](#activity-worker) through and an [Activity Task Queue](#activity-task-queue).
 
-An object that is executed in the client application and receives [activity tasks](#activity-task) from an [activity task queue](#activity-task-queue) it is subscribed to. Once task is received it invokes a correspondent activity.
+- Upon receiving an [Activity Task](#activity-task), an [Activity Worker](#activity-worker) executes the corresponding [Activity](#activity).
 
-### Archival
+### Activity Task Queue
 
-Archival is a feature that automatically moves [histories](#event-history) from persistence to a blobstore after
-the workflow retention period. The purpose of archival is to be able to keep histories as long as needed
-while not overwhelming the persistence store. There are two reasons you may want
-to keep the histories after the retention period has passed:
+[Task Queue](#task-queue) that is used to deliver [Activity Tasks](#activity-task) to [Activity Workers](#activity-worker)
```

Comment:
Are all Task Queues internal? Or just Activity Task Queues? Seems like all of them are?
And what by "internal" - we mean that they are not really accessible or observable?

My understanding is that Task Queues are created in code and tasks are routed to queues in code, but the queues themselves are not "available" outside of the Temporal system - they can't be otherwise managed or observed. Is this correct?

Reply:
I was asked multiple times if we use Kafka for queues. So we want to say that these queues are part of the Temporal service and rely on the same persistence storage as the rest of the Temporal.

Reply:
I think that this is another good thing to condense - Activity Task Queue & Workflow Task Queue -> Task Queue definition. I will put this point there.